packages/dev-middleware/src/inspector-proxy/Device.js

```diff
@@ -97,13 +102,23 @@ export default class Device {
 
   #pagesPollingIntervalId: ReturnType<typeof setInterval>;
 
+  // The device message middleware factory function allowing implementers to handle unsupported CDP messages.
+  #createMessageMiddleware: ?createDeviceMessageMiddleware;
+
+  // The device message middleware instances, per debugger, page, and device connection
+  #messageMiddlewares: WeakMap<
+    DebuggerInfo,
```

Comment:
Also, could we avoid `WeakMap` and structure our data differently to begin with, with proper cleanup when a connection is terminated etc?

Reply:
We could nest the middleware instance under the debugger connection instead. That way, once that is cleared, the middleware should be garbage collected as well. I think a debugger connection always is to a single Page/device, so that should match the scope of the middleware too.

But I probably need some help related to Flow typing on that one, in the previous iteration I put it on the `Page` object but ran into [exact/in-exact typing issues here](https://github.com/facebook/react-native/pull/43291#pullrequestreview-1912645088)

Reply:
✅ This is now nested inside the `this.#debuggerConnection`, and get's cleared once this connection is destroyed.